### PR TITLE
Support for `autonomy analyse dialogues`

### LINKED ISSUE #1581: autonomy analyse should check whether all required dialogues are set
`autonomy analyse` should check for missing dialogues.

### PR DESCRIPTION
**Pull request by angrybayblade**
## Proposed changes

This PR adds support for `autonomy analyse dialogues` command

## Fixes

#1581 

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: autonomy/configurations/constants.py [hunk 1]
```diff
@@ -28,8 +28,6 @@
 DEFAULT_SERVICE_CONFIG_FILE = "service.yaml"
 SERVICE = "service"
 SERVICES = "services"
-HANDLERS = "handlers"
-CLASS_NAME = "class_name"
```
Comment (angrybayblade) [inline autonomy/configurations/constants.py:32]: Moved to `autonomy.analyse.constants`

File: autonomy/analyse/dialogues.py [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+
+        if name_ends_with_dialogue and not class_ends_with_dialogue:
+            raise ValueError(
+                f"Class name for a dialogue should end with `Dialogues`; dialogue={name}; class={class_name}"
+            )
+
+        if class_ends_with_dialogue and not name_ends_with_dialogue:
+            raise ValueError(
+                f"Dialogue name should end with `dialogues`; dialogue={name}; class={class_name}"
+            )
```
Comment (angrybayblade) [inline autonomy/analyse/dialogues.py:73]: Is using a lower case string to perform checks enough or should we make them case sensitive?

File: autonomy/analyse/dialogues.py [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+from pathlib import Path
+from typing import Any, Dict, List, cast
+
+import yaml
+
+from autonomy.analyse.constants import CLASS_NAME, DIALOGUES, DIALOGUES_FILE, MODELS
+
+
+DIALOGUES_LOWER = DIALOGUES.lower()
+
```
Comment (Adamantios) [inline autonomy/analyse/dialogues.py:37]: Isn't this already lowercase?

File: autonomy/analyse/dialogues.py [hunk 4]
```diff
@@ -0,0 +1,106 @@
...
+    loader.exec_module(module)
+
+    return module
+
+
+def validate_and_get_dialogues(configuration: Dict[str, Any]) -> Dict[str, str]:
+    """Returns dialogue names to class name mappings"""
+
+    mapping = {}
+    for name, info in cast(Dict[str, Dict[str, str]], configuration[MODELS]).items():
```
Comment (Adamantios) [inline autonomy/analyse/dialogues.py:56]: No need to pass the full config.

```suggestion
def validate_and_get_dialogues(models_configuration: Dict[str, Dict[str, str]]) -> Dict[str, str]:
    """Returns dialogue names to class name mappings"""

    mapping = {}
    for name, info in models_configuration.items():
```

File: autonomy/analyse/dialogues.py (hunk 2 above)
Comment (Adamantios) [reply autonomy/analyse/dialogues.py:73]: Should be fine, we have linters for checking the naming conventions.

File: autonomy/cli/analyse.py [hunk 5]
```diff
@@ -209,6 +213,43 @@ def run_handler_check(
...
+@pass_ctx
+@click.option(
+    "--dialogue",
+    "-d",
+    "dialogues",
+    type=str,
+    default=[
+        "abci",
+    ],
+    help="Specify which handlers to check. Eg. -h handler_a -h handler_b -h handler_c",
```
Comment (Adamantios) [inline autonomy/cli/analyse.py:226]:
```suggestion
    help="Specify which dialogues to check. Eg. -d dialogue_a -d dialogue_b -d dialogue_c",
```

File: tests/test_autonomy/test_cli/test_analyse/test_check_dialogues.py [hunk 6]
```diff
@@ -0,0 +1,188 @@
...
+            config_data=config_data,
+            dialogues=["AbciDialogues"],
+        )
+
+        with self.patch_list_skills():
+            result = self.run_cli(commands=("-d", "some_dialogues"))
+
+        assert result.exit_code == 1, result.stdout
+        assert "dialogue SomeDialogues declared in" in result.output
+        assert "is missing from" in result.output
```
Comment (Adamantios) [inline tests/test_autonomy/test_cli/test_analyse/test_check_dialogues.py:188]: Lots of code dup. These could be a single parametrized test.

**Review by Adamantios — COMMENTED** [review]
Let's also add the `tox -e check-dialogues` check to the `Makefile`.

File: tests/test_autonomy/test_cli/test_analyse/test_check_dialogues.py (hunk 6 above)
Comment (angrybayblade) [reply tests/test_autonomy/test_cli/test_analyse/test_check_dialogues.py]: I did not parametrise the tests because it makes it hard to debug when there's a failure for on particular edge case, instead I extracted the common logic and automated the test [6f79ed0](https://github.com/valory-xyz/open-autonomy/pull/1786/commits/6f79ed093cca649e377e2c41c3da9445ff116f63)

**Review by Adamantios — APPROVED** [review]

**Review by DavidMinarsch — APPROVED** [review]
